cns/service.go

```diff
@@ -193,27 +210,32 @@ func getTLSConfigFromKeyVault(tlsSettings localtls.TlsSettings, errChan chan<- e
 }
 
 func (service *Service) StartListener(config *common.ServiceConfig) error {
-	log.Debugf("[Azure CNS] Going to start listener: %+v", config)
-
-	// Initialize the listener.
-	if service.Listener != nil {
-		log.Debugf("[Azure CNS] Starting listener: %+v", config)
-		// Start the listener.
-		// continue to listen on the normal endpoint for http traffic, this will be supported
-		// for sometime until partners migrate fully to https
-		if err := service.Listener.Start(config.ErrChan); err != nil {
-			return err
+	log.Debugf("[Azure CNS] Going to start listeners: %+v", config)
+
+	// Initialize listeners.
+	for _, listener := range *service.Listeners { //nolint
+		if &listener != nil {
```

Comment:
Once you make the above changes, this will just be 
```
for _, listener := range service.Listeners {
```
also, we don't check `&listener != nil`, since only listeners that have been successfully initialized previously got added to this list. With that said, we should throw an error when the listener list is empty or don't have the required number of listeners.